src/main/host/descriptor/pipe.rs

```diff
@@ -159,20 +150,16 @@ impl PipeFile {
         // remove any flags that aren't relevant
         let status = self.filter_status(status);
 
-        if do_set_bits {
-            self.status |= status;
-        } else {
-            self.status &= !status;
-        }
+        self.status.set(status, do_set_bits);
```

Comment:
Doesn't this still need to clear the bits if `do_set_bits` is `false`?

Reply:
The documentation for [`set(flags, bool)`](https://docs.rs/bitflags/1.2.1/bitflags/example_generated/struct.Flags.html#method.set) says "Inserts or removes the specified flags depending on the passed value", and I double checked the [source](https://docs.rs/bitflags/1.2.1/src/bitflags/lib.rs.html#698) and it calls `remove(flags)` if it's `false`.

Reply:
Ahh, I missed that `do_set_bits` was getting passed as a parameter there